### PR TITLE
Embed locally-built native image JIT Executor binary on Extended Services

### PR DESCRIPTION
**Pull request by gitgabrio**
@tiagobento @tiagodolphine @ljmotta  

Implement environment variable to embed locally-built jitexecutor-runner

### CODE REVIEW

File: packages/extended-services/env/index.js [hunk 1]
```diff
@@ -34,6 +34,10 @@ module.exports = composeEnv([require("@kie-tools/root-env/env")], {
       default: "21345",
       description: "Extended Services port",
     },
+    EXTENDED_SERVICES__nativeBuildPath_macOS: {
```
Comment (tiagobento) [inline packages/extended-services/env/index.js]: I think `EXTENDED_SERVICES__jitExecutorNativeBinaryPath_macOS` would be a better name.

File: packages/extended-services/package.json [hunk 2]
```diff
@@ -26,7 +26,7 @@
     "build:prod": "rimraf dist jitexecutor && pnpm copy-jitexecutor && pnpm build && pnpm pack-app",
     "build:win32": "make win32",
     "copy-jitexecutor": "run-script-os",
-    "copy-jitexecutor:darwin": "cp ./node_modules/@kie-tools/jitexecutor-native/dist/darwin/jitexecutor jitexecutor && chmod +x jitexecutor",
+    "copy-jitexecutor:darwin": "cp $(build-env extendedServices.jitexecutor.nativeBuildPath.macOS) jitexecutor && chmod +x jitexecutor",
     "copy-jitexecutor:linux": "cp ./node_modules/@kie-tools/jitexecutor-native/dist/linux/jitexecutor jitexecutor && chmod +x jitexecutor",
     "copy-jitexecutor:win32": "COPY /B /Y .\\node_modules\\@kie-tools\\jitexecutor-native\\dist\\win32\\jitexecutor.exe jitexecutor",
```
Comment (tiagobento) [inline packages/extended-services/package.json]: Can you please apply the same strategy for Windows and Linux?

File: packages/extended-services/env/index.js [hunk 3]
```diff
@@ -42,6 +46,11 @@ module.exports = composeEnv([require("@kie-tools/root-env/env")], {
         kieSandboxUrl: getOrDefault(this.vars.EXTENDED_SERVICES__kieSandboxUrl),
         ip: getOrDefault(this.vars.EXTENDED_SERVICES__ip),
         port: getOrDefault(this.vars.EXTENDED_SERVICES__port),
+        jitexecutor: {
+          nativeBuildPath: {
```
Comment (tiagobento) [inline packages/extended-services/env/index.js]: I think `nativeBinaryPath` would be a better name..

**Review by tiagobento — CHANGES_REQUESTED** [review]

File: packages/extended-services/package.json (hunk 2 above)
Comment (gitgabrio) [reply packages/extended-services/package.json]: done

File: packages/extended-services/env/index.js (hunk 1 above)
Comment (gitgabrio) [reply packages/extended-services/env/index.js]: done

File: packages/extended-services/env/index.js (hunk 3 above)
Comment (gitgabrio) [reply packages/extended-services/env/index.js]: done

**Review by yesamer — APPROVED** [review]

**Comment by yesamer** [comment]
@gitgabrio 

The system cannot find the file specified.
 ELIFECYCLE  Command failed with exit code 1.
undefined
D:\a\kie-tools\kie-tools\packages\extended-services:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @kie-tools/extended-services@0.0.0 build:prod: `rimraf dist jitexecutor && pnpm copy-jitexecutor && pnpm build && pnpm pack-app`
Exit status 1
Error: Process completed with exit code 1.

**Comment by tiagobento** [comment]
Yeah, for Windows, the sub-command expression won't work with some extra tweaks. package.json scripts are ran on CMD by default, which doesn't have this syntax available. The solution is to invoke powershell there and change the copy command for the Power Shell equivalent.

**Comment by gitgabrio** [comment]
> Yeah, for Windows, the sub-command expression won't work with some extra tweaks. package.json scripts are ran on CMD by default, which doesn't have this syntax available. The solution is to invoke powershell there and change the copy command for the Power Shell equivalent.

So ?

**Comment by yesamer** [comment]
@gitgabrio not totally sure, but I think you should follow the same approach used in other modules (eg. stunner-editors package.json file)


Screenshot 2022-11-25 at 09 29 12

` change the copy command for the Power Shell equivalent.`
... and understand how to apply this :)

**Comment by tiagobento** [comment]
> > Yeah, for Windows, the sub-command expression won't work with some extra tweaks. package.json scripts are ran on CMD by default, which doesn't have this syntax available. The solution is to invoke powershell there and change the copy command for the Power Shell equivalent.
> 
> 
> 
> So ?
> 
> 

What do you mean "So ?"?

**Comment by gitgabrio** [comment]
> > > Yeah, for Windows, the sub-command expression won't work with some extra tweaks. package.json scripts are ran on CMD by default, which doesn't have this syntax available. The solution is to invoke powershell there and change the copy command for the Power Shell equivalent.
> > 
> > 
> > So ?
> 
> What do you mean "So ?"?

So: what should be done? I've no way to test on windows, and I'm not even experienced with power shell (almost always used linux on my career) - so : what should be done with it ?

**Comment by tiagobento** [comment]
I trust you can find a way to execute a copy command on Power Shell without being experienced with it! 😎

As for testing, you can install a Virtual Machine, or even blindly push a command and the CI will run it for you. You can also do a custom GitHub Action on a repository in your fork that only runs a command on Windows using Power Shell.

**Comment by gitgabrio** [comment]
> @gitgabrio not totally sure, but I think you should follow the same approach used in other modules (eg. stunner-editors package.json file)
> 
> Screenshot 2022-11-25 at 09 29 12
> 
> ` change the copy command for the Power Shell equivalent.` ... and understand how to apply this :)

NOt sure how can I "understand" without trying it. Beside, what I can't follow is that that PR simply replace a value with a variable, and everything is managed inside th ejs/go environment (no pnpm actually involved)
So, replacing the hardcoded path here

`COPY /B /Y .\node_modules\@kie-tools\jitexecutor-native\dist\win32\jitexecutor.exe`
 with a js. variable should not involve anything specific - maybe just the syntax should be tweaked a little bit, but can't test it

**Comment by tiagobento** [comment]
NOTE: You can also use a platform-agnostic tool to do it as well, like 'cpr' or 'copyfiles'. We have plenty of examples of that in the repository.

**Comment by gitgabrio** [comment]
@tiagobento could you pls approve/merge this ? Thanks!

**Review by ljmotta — APPROVED** [review]

**Review by tiagobento — APPROVED** [review]

**Pull request merged**